src/app/server/JointFabricDatastore.h

```diff
@@ -0,0 +1,198 @@
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+
+#include <app-common/zap-generated/cluster-objects.h>
+#include <lib/core/CHIPPersistentStorageDelegate.h>
+#include <lib/core/CHIPVendorIdentifiers.hpp>
+#include <lib/core/NodeId.h>
+#include <vector>
+
+namespace chip {
+namespace app {
+
+/**
+ * A struct which extends the DatastoreNodeInformationEntry type with FriendlyName buffer reservation.
+ */
+struct GenericDatastoreNodeInformationEntry
+    : public Clusters::JointFabricDatastore::Structs::DatastoreNodeInformationEntryStruct::Type
+{
+    GenericDatastoreNodeInformationEntry(NodeId nodeId = 0,
+                                         Clusters::JointFabricDatastore::DatastoreStateEnum state =
+                                             Clusters::JointFabricDatastore::DatastoreStateEnum::kUnknownEnumValue,
+                                         Optional<CharSpan> label = NullOptional)
+    {
+        Set(nodeId, state, label);
+    }
+
+    GenericDatastoreNodeInformationEntry(const GenericDatastoreNodeInformationEntry & op) { *this = op; }
+
+    GenericDatastoreNodeInformationEntry & operator=(const GenericDatastoreNodeInformationEntry & op)
+    {
+        Set(op.nodeID, op.commissioningStatusEntry.state, MakeOptional(op.friendlyName));
+        return *this;
+    }
+
+    void Set(NodeId nodeId, Clusters::JointFabricDatastore::DatastoreStateEnum state, Optional<CharSpan> label = NullOptional)
+    {
+        this->nodeID                         = nodeId;
+        this->commissioningStatusEntry.state = state;
+        Set(label);
+    }
+
+    void Set(Optional<CharSpan> label = NullOptional)
+    {
+        if (label.HasValue())
+        {
+            memset(mFriendlyNameBuffer, 0, sizeof(mFriendlyNameBuffer));
+            if (label.Value().size() > sizeof(mFriendlyNameBuffer))
+            {
+                memcpy(mFriendlyNameBuffer, label.Value().data(), sizeof(mFriendlyNameBuffer));
+                this->friendlyName = CharSpan(mFriendlyNameBuffer, sizeof(mFriendlyNameBuffer));
+            }
+            else
+            {
+                memcpy(mFriendlyNameBuffer, label.Value().data(), label.Value().size());
+                this->friendlyName = CharSpan(mFriendlyNameBuffer, label.Value().size());
+            }
+        }
+        else
+        {
+            this->friendlyName = CharSpan();
+        }
+    }
+
+private:
+    static constexpr size_t kFriendlyNameMaxSize = 32u;
+
+    char mFriendlyNameBuffer[kFriendlyNameMaxSize];
+};
+
+class JointFabricDatastore
```

Comment:
need documentation on usage/purpose.

Iw as looking to understand the include changes that claimed "not intended for embedded devices" ... but I don't have the intent of this class either. So what is it and what does it do?